presto-main/src/test/java/io/prestosql/type/TestTimestampBase.java

```diff
@@ -86,6 +86,10 @@ public void testLiteral()
         assertFunction("TIMESTAMP '2001-1-2 3:4'", createTimestampType(0), sqlTimestampOf(0, 2001, 1, 2, 3, 4, 0, 0, session));
         assertFunction("TIMESTAMP '2001-1-2'", createTimestampType(0), sqlTimestampOf(0, 2001, 1, 2, 0, 0, 0, 0, session));
 
+        assertFunction("TIMESTAMP '123001-01-22 03:04:05.321'", createTimestampType(3), sqlTimestampOf(3, 123001, 1, 22, 3, 4, 5, 321, session));
```

Comment:
Add tests with precisions up to 12. You'll need to move these to BaseTestTimestamp to take advantage of the new assertj-based assertions. E.g.,

```java
assertThat(assertions.expression("TIMESTAMP '2020-05-01 12:34:56.123456789012'"))
        .hasType(createTimestampType(12))
        .isEqualTo(timestamp(12, 2020, 5, 1, 12, 34, 56, 123_456_789_012L));

Reply:
I added more tests in BaseTestTimestamp. I kept the tests here, however.

Also, please add/port any relevant tests to https://github.com/prestosql/presto/pull/3947